frontend/src/metabase/visualizations/components/settings/ChartSettingFieldPicker.styled.tsx

```diff
@@ -1,13 +1,41 @@
 import styled from "@emotion/styled";
 import { color } from "metabase/lib/colors";
 import Icon from "metabase/components/Icon";
+import SelectButton from "metabase/core/components/SelectButton";
+
+export const ChartSettingFieldPickerRoot = styled.div`
+  display: flex;
+  align-items: center;
+  border: 1px solid ${color("border")};
+  border-radius: 0.5rem;
+  padding-right: 1.125rem;
+
+  ${SelectButton.Root} {
+    border: none;
+    padding: 0.75rem;
+  }
+
+  ${SelectButton.Icon} {
+    margin-left: 0;
+    color: ${color("text-dark")};
+    height: 0.625rem;
+  }
+
+  ${SelectButton.Content} {
+    font-size: 0.75rem;
+    line-height: 1rem;
+    margin-right: 0.25rem;
+  }
+`;
 
 export const SettingsIcon = styled(Icon)`
   margin-left: 0.5rem;
   color: ${color("text-medium")};
   cursor: pointer;
   visibility: ${props => (props.onClick ? "visible" : "hidden")};
 
+  display: ${props => (props.onClick ? "block" : "none")};
```

Comment:
Likely don't need both display and visibility here, but I wasn't sure

Reply:
This seems necessary unless you don't render the `<SettingsIcon />` entirely when it shouldn't be displayed, otherwise, it will still take space in the DOM. e.g.

by doing this

```ts
      {onRemove && (
        <SettingsIcon
          data-testid={`remove-${value}`}
          name="close"
          onClick={onRemove}
        />
      )}
```
instead of this
https://github.com/metabase/metabase/blob/ff67d84c113b31286b80465a8ccf1fb1da80cbbb/frontend/src/metabase/visualizations/components/settings/ChartSettingFieldPicker.jsx#L55-L59

--- 

You can also see how the other `<SettingsIcon />` is rendered https://github.com/metabase/metabase/blob/ff67d84c113b31286b80465a8ccf1fb1da80cbbb/frontend/src/metabase/visualizations/components/settings/ChartSettingFieldPicker.jsx#L42-L54

Reply:
This is likely a better approach. I was worried it might cause some tests that are looking for the element to fail, but if that happens I'll just fix the tests lol

Reply:
... it did. Damnit haha